starlette/endpoints.py

```diff
@@ -23,7 +23,15 @@ def __await__(self) -> typing.Generator:
 
     async def dispatch(self) -> None:
         request = Request(self.scope, receive=self.receive)
-        handler_name = "get" if request.method == "HEAD" else request.method.lower()
+
+        if request.method == "HEAD":
+            if hasattr(self, "head"):
+                handler_name = "head"
+            else:
+                handler_name = "get"
+        else:
+            handler_name = request.method.lower()
+
```

Comment:
```suggestion
```suggestion
        method = request.method.lower()
        handler_name = (
            "get" if method == "head" and not hasattr(self, "head") else method
        )
```
```